tools/syz-execprog/execprog.go

```diff
@@ -418,29 +415,68 @@ func createConfig(target *prog.Target, features *host.Features, featuresFlags cs
 		}
 		execOpts.ExecFlags |= ipc.FlagCollectComps
 	}
-	execOpts.EnvFlags |= ipc.FeaturesToFlags(features.ToFlatRPC(), featuresFlags)
-	return config, execOpts
-}
-
-func buildCallList(target *prog.Target, enabled []string) map[*prog.Syscall]bool {
-	syscallsIDs, err := mgrconfig.ParseEnabledSyscalls(target, enabled, nil)
+	cfg := &mgrconfig.Config{
+		Sandbox:    ipc.FlagsToSandbox(execOpts.EnvFlags),
+		SandboxArg: execOpts.SandboxArg,
+		Derived: mgrconfig.Derived{
+			TargetOS:     target.OS,
+			TargetArch:   target.Arch,
+			TargetVMArch: target.Arch,
+			Target:       target,
+			SysTarget:    targets.Get(target.OS, target.Arch),
+			Syscalls:     syscalls,
+		},
+	}
+	checker := vminfo.New(cfg)
+	files, requests := checker.StartCheck()
+	fileInfos := host.ReadFiles(files)
+	featureInfos, err := host.SetupFeatures(target, config.Executor, flatrpc.AllFeatures, featuresFlags)
 	if err != nil {
-		log.Fatalf("failed to parse enabled syscalls: %v", err)
-	}
-	enabledSyscalls := make(map[*prog.Syscall]bool)
-	for _, id := range syscallsIDs {
-		enabledSyscalls[target.Syscalls[id]] = true
+		log.Fatal(err)
 	}
-	calls, disabled, err := host.DetectSupportedSyscalls(target, "none", enabledSyscalls)
+	env, err := ipc.MakeEnv(config, 0)
 	if err != nil {
-		log.Fatalf("failed to detect host supported syscalls: %v", err)
+		log.Fatalf("failed to create ipc env: %v", err)
 	}
-	for c, reason := range disabled {
-		log.Logf(0, "unsupported syscall: %v: %v", c.Name, reason)
+	defer env.Close()
+	var results []rpctype.ExecutionResult
```

Comment:
Agree. I bit of brute-forced things here. Did not want to completely refactor execprog as part of this PR. As it's large each part won't get very detailed review.

Also I assume it all will need to change if/when we merge fuzzer+executor and make it all C++. Then execprog will need to talk flatrpc with executor. But then it will be able to use all of it's flexibility (start executor with N procs and ask to run programs used during check procedure and actual programs from the log).